pkg/nodebootstrap/managed_al2.go

```diff
@@ -136,5 +137,22 @@ func createMimeMessage(writer io.Writer, scripts, cloudboots []string, mimeBound
 			return err
 		}
 	}
+
+	if nodeConfig != nil {
+		yamlData, err := yaml.Marshal(nodeConfig)
+		if err != nil {
+			return fmt.Errorf("error marshalling node configuration: %w", err)
+		}
+		part, err := mw.CreatePart(map[string][]string{
+			"Content-Type": {"application/node.eks.aws"},
+		})
+		if err != nil {
+			return err
+		}
+		if _, err = part.Write(yamlData); err != nil {
+			return err
+		}
+	}
+
 	return mw.Close()
```

Comment:
Maybe we can just `defer mw.Close()` at the beginning of the function?

Reply:
I agree that it' unnecessary' given that this variable is local, however, i think it's generally considered a good Go practice to consistently defer Close() calls for reasons like consistency and clarity and minimizing unexpected behaviour.

Reply:
This is non-blocking and can be addressed separately another day, maybe for all similar cases.